crates/storage/provider/src/either_writer.rs

```diff
@@ -109,6 +114,42 @@ impl<'a> EitherWriter<'a, (), ()> {
             ))
         }
     }
+
+    /// Creates a new [`EitherWriter`] for storages history based on storage settings.
+    #[cfg(all(unix, feature = "rocksdb"))]
+    pub fn new_storages_history<P>(
+        provider: &P,
+        rocksdb: RocksDBProvider,
+    ) -> ProviderResult<EitherWriterTy<'a, P, tables::StoragesHistory>>
+    where
+        P: DBProvider + NodePrimitivesProvider + StorageSettingsCache,
+        P::Tx: DbTxMut,
+    {
+        if provider.cached_storage_settings().storages_history_in_rocksdb {
+            Ok(EitherWriter::RocksDB(rocksdb))
+        } else {
+            Ok(EitherWriter::Database(provider.tx_ref().cursor_write::<tables::StoragesHistory>()?))
+        }
+    }
+
+    /// Creates a new [`EitherWriter`] for transaction hash numbers based on storage settings.
+    #[cfg(all(unix, feature = "rocksdb"))]
+    pub fn new_transaction_hash_numbers<P>(
```

Comment:
is this what you mean by constructor? @Rjected

Reply:
Done\! Introduced type aliases (`RocksTxArg`, `RocksTxRefArg`) that resolve to `RocksTx<'a>` or `&'a RocksTx<'a>` when rocksdb is enabled, and `()` when disabled. The constructors are now always available - internal cfg blocks handle the routing.